### ISSUE TITLE
Be more explicit about what a copy is

### ISSUE THREAD

**Issue by JamesMcMahon** [jashkenas/lil-license#6]
The line from the license,

> This license, or a link to its text, must be included with all copies of the software and any derivative works.

is a little ambiguous, and is most likely based on this line from the MIT license,

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

The BSD license has much clearer verbiage around what a copy is.

>  * Redistributions of source code must retain the above copyright
      notice, this list of conditions and the following disclaimer.
    
>   * Redistributions in binary form must reproduce the above copyright
      notice, this list of conditions and the following disclaimer in the
      documentation and/or other materials provided with the distribution.

I'd also note that I think you are critically missing a call to include the copyright, but that may be what is meant as "the license".

**Comment by aprescott** [comment]
I came to the list of issues for this project with similar concerns. Particularly around maintaining the copyright attribution notice as part of the requirements.

**Comment by matoous** [comment]
Is it ok for someone to just replace this line with BSD license lines and use it as a license for their project?

**Comment by JamesMcMahon** [comment]
You can always use a custom license on a project but you lose all the advantages that come with a standard license including legal vetting and industry acceptance. In general it’s a bad call to roll your own.

**Comment by jashkenas** [comment]
Yes — the hope here is to boil this down to the essentials, vet it again, submit to the relevant standards bodies, then build industry acceptance.

"This license" refers to the entire document: The Lil License. That includes your copyright notice, the grant of permissions, the limiting conditions and the warranty disclaimer.

I think it's actually much clearer to refer to "this license" and not explicitly split out "the above copyright notice" and "this permission notice", because then people try to think of the unmentioned bits that they can leave out. Under MIT, can your copy omit the warranty disclaimer because it's not explicitly mentioned?

Better to refer to the whole: "this license".

**Comment by aprescott** [comment]
For what it's worth, as a developer looking at the license terms, when I read "this license", the meaning was/is not at all clear to me. The phrase "this license" can refer to the completely general form of the license as available with a placeholder copyright notice. What makes it extra ambiguous is that you can merely _link_ to a copy of the license. Which strongly suggests to someone distributing a copy or a derivative work that they could do something like this:

```js
/* foobar.js, licensed under the Lil License v1: http://lillicense.org/v1.html */
```

That doesn't include a copyright attribution notice, but there is an included a link to the copy of the license in its general form. (I imagine this sort of ambiguity is why there's a distinction made between an immutable general version of a license's text and an accompanying copyright notice. There is a similar unfortunate confusion with how people use Apache v2 and its Appendix, versus using a NOTICE file.)

I would personally be quite hesitant to replace something like MIT with the Lil License without clearer/stronger language around the copyright notice.

**Comment by jashkenas** [comment]
That's a great point, about `http://lillicense.org/v1.html` not being an acceptable place to link to, while at the same time obviously inviting that sort of linking behavior.

What would you prefer as the ideal fix for this? Take the authors out of the license text, and put in an AUTHORS file? Get rid of the "or a link to its text" provision?

Or perhaps an example of how to link with copyright inline:

```
// Copyright (c) Jane Smith 2017, licensed under the Lil License v1: http://lillicense.org/v1.html
```

**Comment by aprescott** [comment]
(IANAL, for all this, by the way, I'm just coming at this from a developer perspective.)

Personally I think a separate AUTHORS file is a bit too costly and unnecessary for most people. From a developer-friendliness perspective, I think the ability to merely link (and not have to distribute along other notices, like LICENSE, or AUTHORS) is good, particularly for stuff like JS minification and single-file bundling/concatenation. I'm in no position to mandate solutions, so it all just depends on what you want to require and allow people to do, of course.

If requiring copyright notice attribution is a goal, ultimately I think it needs to come with clearer language. As an example, if I, as someone distributing licensed software, understand from the license terms that I have to include, say, somewhere with the software (a) the copyright notice as attribution, and (b) the terms under which that software is available (by link, reference, or some other mechanism), then I can easily check if I'm clearing the minimum bar.

Beyond MIT as an example, the copyleft MPL 2.0 sections 3.1 and 3.4, etc., try to get at this sort of thing without being too restrictive. There too, "this License" seems to me to refer to the general text of the license, in contrast to a specific copyright notice + license pairing. Trying to cover both ideas with one single phrase seems to be a bit too ambiguous.

**Comment by aprescott** [comment]
Edited for s/license/software/ in a few places, sorry about that. :)
